projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/pojo/AnalysisTest.java

```diff
@@ -0,0 +1,125 @@
+package org.batfish.datamodel.pojo;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.testing.EqualsTester;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.batfish.common.BatfishException;
+import org.batfish.datamodel.questions.Question;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link Analysis}. */
+@RunWith(JUnit4.class)
+public class AnalysisTest {
+
+  @Rule public ExpectedException _thrown = ExpectedException.none();
+
+  private class TestQuestion extends Question {
+    String _name;
+
+    TestQuestion(String name) {
+      this._name = name;
+    }
+
+    @Override
+    public boolean getDataPlane() {
+      return false;
+    }
+
+    @Override
+    public String getName() {
+      return _name;
+    }
+
+    @Override
+    public boolean getTraffic() {
+      return false;
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(TestQuestion.class).add("name", _name).toString();
+    }
+  }
+
+  @Test
+  public void testConstructorWithName() {
+    Analysis a = new Analysis("analysis");
+    assertThat(a.getName(), equalTo("analysis"));
+    assertTrue(a.getQuestions().isEmpty());
+  }
+
+  @Test
+  public void testGetterSetter() {
```

Comment:
I think this function is doing two things, testing the Constructor with names and questions and testing the setters and getters. The name testGetterSetter seams too general for the purpose